// carousel/index.js
(function() {
  var Carousel, domready;

  domready = require('domready');

  module.exports = Carousel = (function() {
    function Carousel(selector) {
      this.selector = selector;
      if (!this.selector) {
        console.error("Please specify a CSS selector when creating a new Carousel, e.g. new Carousel('#my-carousel')");
      }
      domready((function(_this) {
        return function() {
          _this.slide_count = document.querySelectorAll(_this.selector + " > *").length;
          document.body.addEventListener('keydown', function(event) {
            var ref, ref1;
            if ((ref = event.keyCode) === 39 || ref === 40) {
              _this.next();
            }
            if ((ref1 = event.keyCode) === 37 || ref1 === 38) {
              return _this.prev();
            }
          });
          return _this.getOffset();
        };
      })(this));
    }

    Carousel.prototype.getOffset = function() {
      this.offset = parseInt(document.body.scrollTop / window.innerHeight);
      if ((document.body.scrollTop % window.innerHeight) > window.innerHeight / 2) {
        return this.offset++;
      }
    };

    Carousel.prototype.next = function() {
      this.getOffset();
      if (this.offset !== this.slide_count - 1) {
        this.offset++;
      }
      return this.animate();
    };

    Carousel.prototype.prev = function() {
      this.getOffset();
      if (this.offset !== 0) {
        this.offset--;
      }
      return this.animate();
    };

    Carousel.prototype.animate = function() {
      if (typeof jQuery !== "undefined") {
        return $("html, body").animate({
          scrollTop: window.innerHeight * this.offset
        });
      } else {
        return document.body.scrollTop = window.innerHeight * this.offset;
      }
    };

    return Carousel;

  })();

}).call(this);
